lib/posix/options/Kconfig.pthread

```
@@ -156,6 +156,10 @@ config POSIX_THREAD_PRIO_PROTECT
 
 config POSIX_THREAD_SAFE_FUNCTIONS
 	bool "POSIX thread-safe functions"
+	select COMMON_LIBC_ASCTIME_R
+	select COMMON_LIBC_GMTIME_R
+	select COMMON_LIBC_LOCALTIME_R_UTC
+	select COMMON_LIBC_CTIME_R
 	help
 	  Select 'y' here to enable POSIX thread-safe functions including asctime_r(), ctime_r(),
 	  flockfile(), ftrylockfile(), funlockfile(), getc_unlocked(), getchar_unlocked(),
```

Darn - I just realized `_POSIX_THREAD_SAFE_FUNCTIONS` is an Option, and not an Option Group :(

Based on (posixoptions(7))[https://man7.org/linux/man-pages/man7/posixoptions.7.html], it seems to be a combination of  `POSIX_C_LANG_SUPPORT_R` and `POSIX_FILE_SYSTEM_R` Option Groups

For reference
https://pubs.opengroup.org/onlinepubs/9799919799/xrat/V4_subprofiles.html

So likely we will need a new `Kconfig.c_lang_r` like this, approximately.

```
config POSIX_C_LANG_SUPPORT_R
    select COMMON_LIBC_ASCTIME_R
    select COMMON_LIBC_CTIME_R
    select COMMON_LIBC_GMTIME_R
    select COMMON_LIBC_LOCALTIME_R
```

With that, I think there are some nuances with how `POSIX_THREAD_SAFE_FUNCTIONS` should behave. Let me know if this makes sense.
```suggestion
config POSIX_THREAD_SAFE_FUNCTIONS
	bool "POSIX thread-safe functions"
	select POSIX_FILE_SYSTEM_R if POSIX_FILE_SYSTEM
	select POSIX_C_LANG_SUPPORT_R
	help
	  Select 'y' here to enable POSIX thread-safe functions including asctime_r(), ctime_r(),
	  flockfile(), ftrylockfile(), funlockfile(), getc_unlocked(), getchar_unlocked(),
```